modules/azure-firewall/diagnostics.tf

```diff
@@ -0,0 +1,39 @@
+resource "azurerm_log_analytics_workspace" "main" {
+  count               = length(var.firewall)
+  name                = "${var.project}-${var.env}-fw-loganalytics"
+  location            = azurerm_resource_group.main[count.index].location
+  resource_group_name = azurerm_resource_group.main[count.index].name
+  sku                 = "PerGB2018"
+  retention_in_days   = 30
+}
+
+resource "azurerm_monitor_diagnostic_setting" "firewall_diagnostics" {
+  count                      = length(var.firewall)
+  name                       = "fw-log-analytics"
+  target_resource_id         = azurerm_firewall.main[count.index].id
+  log_analytics_workspace_id = azurerm_log_analytics_workspace.main[count.index].id
+
+  log {
```

Comment:
you can use something like this to enable all available diagnostics without having to know what is available:
https://github.com/hmcts/cnp-aks-rbac-pipelines/blob/master/terraform/app-gateway-module/main.tf#L122-L124
https://github.com/hmcts/cnp-aks-rbac-pipelines/blob/master/terraform/app-gateway-module/main.tf#L137-L163